spec/models/dynamic_field_spec.rb

```diff
@@ -0,0 +1,186 @@
+require 'rails_helper'
+
+RSpec.describe DynamicField, type: :model do
+  describe '#new' do
+    context 'when parameters are correct' do
+      subject { FactoryBot.create(:dynamic_field) }
+
+      it { is_expected.to be_a DynamicField }
+
+      its(:string_key) { is_expected.to eql 'term' }
+      its(:display_label) { is_expected.to eql 'Value' }
+      its(:field_type) { is_expected.to eql DynamicField::Type::CONTROLLED_TERM }
+      its(:sort_order) { is_expected.to be 7 }
+      its(:filter_label) { is_expected.to eql 'Name' }
+      its(:is_facetable) { is_expected.to eql true }
+      its(:controlled_vocabulary) { is_expected.to eql 'name_role' }
+      its(:select_options) { is_expected.to be_nil }
+      its(:is_keyword_searchable) { is_expected.to be false }
+      its(:is_title_searchable) { is_expected.to be false }
+      its(:is_identifier_searchable) { is_expected.to be false }
+      its(:dynamic_field_group) { is_expected.to be_a DynamicFieldGroup }
+    end
+
+    context 'when missing string_key' do
+      let(:dynamic_field) { FactoryBot.build(:dynamic_field, string_key: nil) }
+
+      it 'does not save' do
+        expect(dynamic_field.save).to be false
+      end
+
+      it 'returns correct error' do
+        dynamic_field.save
+        expect(dynamic_field.errors.full_messages).to include 'String key can\'t be blank'
```

Comment:
Optional change: For readability, I tend to prefer double quotes when a single quote is in a string, even when we're not interpolating anything.  Not sure if rubocop would complain about that though.

Reply:
Since its just one quotation mark, i usually don't switch it to double quotation mark.